platforms/core-configuration/core-serialization-codecs/src/main/kotlin/org/gradle/internal/serialize/codecs/core/TaskNodeCodec.kt

```diff
@@ -230,11 +230,12 @@ suspend fun <T> T.withTaskOf(
     action: suspend () -> Unit
 ) where T : IsolateContext, T : MutableIsolateContext {
     withIsolate(IsolateOwners.OwnerTask(task), codec) {
-        withPropertyTrace(PropertyTrace.Task(taskType, task.identityPath.path)) {
+        val taskTrace = PropertyTrace.Task(taskType, task.identityPath.path)
+        withPropertyTrace(taskTrace) {
             if (task.isCompatibleWithConfigurationCache) {
                 action()
             } else {
-                forIncompatibleType(task.identityPath.path, action)
+                forIncompatibleTask(taskTrace, task.reasonTaskIsIncompatibleWithConfigurationCache.get(), action)
```

Comment:
💅 The current `trace` can be accessed from the context via the `trace` property, no need to declare a variable to hold the taskTrace

Reply:
I wondered whether a reentrant call could set a different trace, but I can't really see now how that would happen